metricbeat/module/docker/diskio/helper.go

```diff
@@ -45,56 +50,64 @@ func (io *BLkioService) getBlkioStatsList(rawStats []docker.Stat, dedot bool) []
 	return formattedStats
 }
 
-func (io *BLkioService) getBlkioStats(myRawStat *docker.Stat, dedot bool) BlkioStats {
+// TODO: Use StorageStats instead of BlkioStats on Windows
+func (io *BlkioService) getBlkioStats(myRawStat *docker.Stat, dedot bool) BlkioStats {
 	newBlkioStats := io.getNewStats(myRawStat.Stats.Read, myRawStat.Stats.BlkioStats.IoServicedRecursive)
-	oldBlkioStats, exist := io.BlkioSTatsPerContainer[myRawStat.Container.ID]
+	bytesBlkioStats := io.getNewStats(myRawStat.Stats.Read, myRawStat.Stats.BlkioStats.IoServiceBytesRecursive)
 
 	myBlkioStats := BlkioStats{
 		Time:      myRawStat.Stats.Read,
 		Container: docker.NewContainer(myRawStat.Container, dedot),
+
+		serviced:      newBlkioStats,
+		servicedBytes: bytesBlkioStats,
 	}
 
+	oldBlkioStats, exist := io.BlkioStatsPerContainer[myRawStat.Container.ID]
 	if exist {
 		myBlkioStats.reads = io.getReadPs(&oldBlkioStats, &newBlkioStats)
 		myBlkioStats.writes = io.getWritePs(&oldBlkioStats, &newBlkioStats)
 		myBlkioStats.totals = io.getTotalPs(&oldBlkioStats, &newBlkioStats)
 	}
 
-	io.BlkioSTatsPerContainer[myRawStat.Container.ID] = newBlkioStats
+	// FIXME: Memory leak
+	io.BlkioStatsPerContainer[myRawStat.Container.ID] = newBlkioStats
```

Comment:
I think we have a memory leak here, entries are never removed, I'll take a look to this too.